### PR TITLE
22 build summary detail amt

### LINKED ISSUE #22: BUILD: summary detail

### PR DESCRIPTION
**Pull request by allison-truhlar**
### What changes did I make?
- Added the SummaryDetail component and associated child components (e.g., ingredients, instructions)
- Added BackgroundBlur component that appears behind SummaryDetail. Also, if you click on it, the SummaryDetail will close.
- Added basic Tailwind styling to the SummaryDetail component and associated child components

### Why did I make these changes?
- When the user clicks on "view recipe details," the project requires that a Summary Detail component with the detailed recipe information appear.

### Known issues
I'm opening this PR with a few known issues because I'm stuck on them, and they're more styling-related, so I'm going to open a new issue for styling this component and add these items in the issue:
- The close button in SummaryDetail refuses to stay in the upper right corner! I had it working and then something must have changed in the code when I was working on other thing because it no longer stays.
- SummaryDetail does not "slide" in and out, despite my best attempts with the CSS. It just appears/disappears.
- I can't get the picture at the top of SummaryDetail to spread across the entire width of the component and zoom in on the image, without increasing the height of the div it's contained in. If it fills the width, it often exceeds the height of the div and overflows into the next div.

### Screenshots

<details><summary>Top of the SummaryDetail component</summary>

Screenshot 2023-10-27 at 2 26 50 PM

</details> 

<details><summary>Bottom of the SummaryDetail component (a little bit of the middle is cutoff between the two images)</summary>

Screenshot 2023-10-27 at 2 26 57 PM

</details>

### CODE REVIEW

File: src/components/summaryDetail/Ingredients.jsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { nanoid } from 'nanoid';
+
+export default function Ingredients({ sections }) {
+  const ingredientEls = sections.map((section) => section.components.map((component) => (<p key={nanoid()} className="pb-2 text-xs">{component.raw_text}</p>)));
+
```
Comment (komalgill0310) [inline src/components/summaryDetail/Ingredients.jsx:4]: As Robester mentioned in the code review of my code, "I assume nanoid is a random key generator. It is not recommended to use a random key generator as a key. You should use some sort of combination of strings from the recipes object or a unique id provided by the backend, which should also be in the recipes object."

File: src/components/backgroundBlur/BackgroundBlur.jsx [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import "./BackgroundBlur.css"
+
+export default function backgroundBlur({handleSummaryDetailClose}) {
```
Comment (komalgill0310) [inline src/components/backgroundBlur/BackgroundBlur.jsx]: Function names should start with Capital Letters.

File: src/components/summaryDetail/SummaryDetail.jsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import './SummaryDetail.css';
+import { FiChevronUp, FiChevronDown, FiVideo } from 'react-icons/fi';
+
+import Ingredients from './Ingredients';
+import Nutrition from './Nutrition';
+import Instructions from './Instructions';
+import Categories from './Categories';
+
+export default function SummaryDetail({
+  recipeDetail, setRecipeDetail, isDetailShown, toggleIsDetailShown, handleSummaryDetailClose,
```
Comment (komalgill0310) [inline src/components/summaryDetail/SummaryDetail.jsx]: 1. setRecipeDetail and toggleIsDetailShown are not being used in the logic, so I guess there is no need for them to pass as a function parameter.

File: src/components/summaryDetail/Nutrition.jsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  }
+
+  if (Object.keys(nutrition).length === 0) {
+    return nutritionEls = <p text-xs>No nutrition data available</p>;
+  }
+  nutritionEls = Object.entries(nutrition)
+    .filter(([key, value]) => key !== 'updated_at')
+    .sort(([keyA], [keyB]) => keyA.localeCompare(keyB))
+    .map(([key, value]) => (
+      <p key={nanoid()} className="border-t text-xs">
```
Comment (komalgill0310) [inline src/components/summaryDetail/Nutrition.jsx]: As Robester mentioned in the code review of my code, "I assume nanoid is a random key generator. It is not recommended to use a random key generator as a key. You should use some sort of combination of strings from the recipes object or a unique id provided by the backend, which should also be in the recipes object."

File: src/components/summaryDetail/Instructions.jsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import { nanoid } from 'nanoid';
+
+export default function Instructions({ instructions }) {
+  const instructionEls = instructions.map((instruction, index) => (
+    <p key={nanoid()} className="pt-4 text-xs">
```
Comment (komalgill0310) [inline src/components/summaryDetail/Instructions.jsx]: As Robester mentioned in the code review of my code, "I assume nanoid is a random key generator. It is not recommended to use a random key generator as a key. You should use some sort of combination of strings from the recipes object or a unique id provided by the backend, which should also be in the recipes object."

File: src/components/summaryDetail/Categories.jsx [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import { nanoid } from 'nanoid';
```
Comment (komalgill0310) [inline src/components/summaryDetail/Categories.jsx]: this import is unnecessary as the logic does not use nanoid.

**Review by komalgill0310 — CHANGES_REQUESTED** [review]
I will discuss the CSS changes with @AOA19 and the known issues.

File: src/components/summaryDetail/Ingredients.jsx (hunk 1 above)
Comment (allison-truhlar) [reply src/components/summaryDetail/Ingredients.jsx:4]: Addressed in commit [9104615](https://github.com/chingu-voyages/v46-tier2-team-20/pull/51/commits/9104615e94971216a6810332980e97b2eacc012a), below

File: src/components/backgroundBlur/BackgroundBlur.jsx (hunk 2 above)
Comment (allison-truhlar) [reply src/components/backgroundBlur/BackgroundBlur.jsx]: Resolved with commit [997c989](https://github.com/chingu-voyages/v46-tier2-team-20/pull/51/commits/997c98971aaf211d9e518c3dc124781e917bcad0), below

File: src/components/summaryDetail/SummaryDetail.jsx (hunk 3 above)
Comment (allison-truhlar) [reply src/components/summaryDetail/SummaryDetail.jsx]: Addressed in commit [8eb5dbd](https://github.com/chingu-voyages/v46-tier2-team-20/pull/51/commits/8eb5dbdf846190aaed62d1d8c0df0ec0033b39a5), below

File: src/components/summaryDetail/Nutrition.jsx (hunk 4 above)
Comment (allison-truhlar) [reply src/components/summaryDetail/Nutrition.jsx]: Addressed with commit [6148bb1](https://github.com/chingu-voyages/v46-tier2-team-20/pull/51/commits/6148bb1f4f35ee60f226aa0f13b608da3f5975ad), below

File: src/components/summaryDetail/Instructions.jsx (hunk 5 above)
Comment (allison-truhlar) [reply src/components/summaryDetail/Instructions.jsx]: Addressed with commit [919f74e](https://github.com/chingu-voyages/v46-tier2-team-20/pull/51/commits/919f74e0503b9e9cbe3044503f0b9f3e9f13b4f4)

File: src/components/summaryDetail/Categories.jsx (hunk 6 above)
Comment (allison-truhlar) [reply src/components/summaryDetail/Categories.jsx]: Addressed with commit [a118fed](https://github.com/chingu-voyages/v46-tier2-team-20/pull/51/commits/a118fed58e19eaa879bff38aa690a26fb0bbcc39)

**Comment by allison-truhlar** [comment]
Hi @komalgill0310! Thank you for your review! I made all the changes you suggested - I included a comment after each of your comments that indicates which commit you can find the change in. Please let me know if you have any more changes to suggest!

**Review by komalgill0310 — APPROVED** [review]

**Pull request merged**